### PR TITLE
[BD-14] Add Content Libraries v2 authoring MFE

### PR DESCRIPTION
**Pull request by arbrandes**
Adds the Content Libraries v2 MFE to the devstack.

See https://github.com/open-craft/frontend-app-library-authoring/pull/2 for testing instructions.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @arbrandes! I've created [BLENDED-428](https://openedx.atlassian.net/browse/BLENDED-428) to keep track of it in Jira.  More details are on the [BD-14](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1545011241/BD-14+Blockstore+Powered+Content+Libraries+Taxonomies) project page.

When this pull request is ready, tag your edX technical lead.

File: options.mk [hunk 1]
```diff
@@ -38,7 +38,7 @@ FS_SYNC_STRATEGY ?= local-mounts
 # TODO: Re-evaluate this list and consider paring it down to a tighter core.
 #       The current value was chosen such that it would not change the existing
 #       Devstack behavior.
-DEFAULT_SERVICES ?= lms+studio+ecommerce+discovery+credentials+forum+edx_notes_api+registrar+gradebook+program-console+frontend-app-publisher
+DEFAULT_SERVICES ?= lms+studio+ecommerce+discovery+credentials+forum+edx_notes_api+registrar+gradebook+program-console+frontend-app-publisher+frontend-app-library-authoring
```
Comment (kdmccormick) [inline options.mk]: I recommend leaving the new MFE out of the default services list until it becomes something that Open edX devs are likely to need by default. We're doing that with `frontend-app-learning` currently.

You'd still be able to start it with `make dev.up.frontend-app-library-authoring`.

File: docker-compose.yml [hunk 2]
```diff
@@ -489,6 +489,21 @@ services:
...
+    working_dir: '/edx/app/frontend-app-library-authoring'
+    container_name: "edx.${COMPOSE_PROJECT_NAME:-devstack}.frontend-app-library-authoring"
+    networks:
+      default:
+        aliases:
+          - edx.devstack.frontend-app-library-authoring
+    ports:
+      - "3001:3001"
+    depends_on:
+      - lms
```
Comment (kdmccormick) [inline docker-compose.yml:522]: It depends on `studio` too, right?

**Comment by mavidser** [comment]
Code LGTM, please add details of the MFE to the `README.md`, wherever relevant (MFE urls section, Useful Commands, etc)

**Comment by arbrandes** [comment]
@kdmccormick, can you take a look at this as well?

File: repo.sh [hunk 3]
```diff
@@ -36,6 +36,7 @@ repos=(
 
 non_release_repos=(
     "https://github.com/edx/frontend-app-learning.git"
+    "https://github.com/edx/frontend-app-library-authoring.git"
```
Comment (kdmccormick) [inline repo.sh:40]: This is obviously dependent on us moving the repo to the `edx` org -- I'll make sure that happens soon.

File: docker-sync.yml [hunk 4]
```diff
@@ -53,3 +53,8 @@ syncs:
     host_disk_mount_mode: 'cached'
     src: '../src/'
     sync_excludes: [ '.git', '.idea' ]
+
+  frontend-app-library-authoring-sync:
+    host_disk_mount_mode: 'cached'
+    src: '../frontend-app-library-authoring/'
+    sync_excludes: [ '.git', '.idea' ]
```
Comment (kdmccormick) [inline docker-sync.yml:65]: Can you also make the corresponding change in `docker-compose-sync.yml`?

**Review by kdmccormick — COMMENTED** [review]
Thanks for the change!

I introduced some merge conflicts with my most recent commit to devstack -- hopefully those should be easy to resolve.

Once you rebase, could you update the new [service list in the README](https://github.com/edx/devstack#service-list)?

**Comment by arbrandes** [comment]
@kdmccormick, thanks for the comments here, once more.  I'll tackle them this week.  The rebase shouldn't be a big deal, though thanks for the heads-up!

File: repo.sh (hunk 3 above)
Comment (kdmccormick) [reply repo.sh:40]: (Done)

**Review by kdmccormick — APPROVED** [review]
Nice; looks good! Of course, it won't actually work until your basic functionality PR is merged to https://github.com/edx/frontend-app-library-authoring/. As soon as that merges, I'll merge this too.

File: options.mk (hunk 1 above)
Comment (arbrandes) [reply options.mk]: Agreed, and done.

File: docker-compose.yml (hunk 2 above)
Comment (arbrandes) [reply docker-compose.yml:522]: Nice catch.  Fixed.

File: docker-sync.yml (hunk 4 above)
Comment (arbrandes) [reply docker-sync.yml:65]: Done.

File: options.mk [hunk 5]
```diff
@@ -74,7 +74,7 @@ credentials+discovery+ecommerce+edx_notes_api+forum+frontend-app-publisher+front
 # Separated by plus signs.
 # Separated by plus signs. Listed in alphabetical order for clarity.
 EDX_SERVICES ?= \
-analyticspipeline+credentials+discovery+ecommerce+edx_notes_api+forum+frontend-app-learning+frontend-app-publisher+gradebook+lms+lms_watcher+marketing+program-console+registrar+registrar-worker+studio+studio_watcher+xqueue+xqueue_consumer
+analyticspipeline+credentials+discovery+ecommerce+edx_notes_api+forum+frontend-app-learning+frontend-app-publisher+frontend-app-library-authoring+gradebook+lms+lms_watcher+marketing+program-console+registrar+registrar-worker+studio+studio_watcher+xqueue+xqueue_consumer
```
Comment (kdmccormick) [inline options.mk]: (the tiniest of nits -- an alphabetization fix)
```suggestion
analyticspipeline+credentials+discovery+ecommerce+edx_notes_api+forum+frontend-app-learning+frontend-app-library-authoring+frontend-app-publisher+gradebook+lms+lms_watcher+marketing+program-console+registrar+registrar-worker+studio+studio_watcher+xqueue+xqueue_consumer
```

Comment (arbrandes) [reply options.mk]: Done!

**Comment by arbrandes** [comment]
@kdmccormick, now that the basic MFE is up, this should also be good to merge (if it doesn't trip Travis, that is).

**Comment by kdmccormick** [comment]
Looks good! The flakiness in the ecommerce shard is a known issue, which we can ignore.

**Pull request merged**